Sources/StreamChat/Utils/TextLinkDetector.swift

```diff
@@ -0,0 +1,57 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+/// The url and position of a link in a text.
+public struct TextLink: Equatable {
+    public let url: URL
+    public let range: NSRange
+}
+
+/// An object responsible to detect and parse links in text.
+public class TextLinkDetector {
+    private let detector: NSDataDetector?
+
+    public init() {
+        let types: NSTextCheckingResult.CheckingType = [.link]
+        do {
+            detector = try NSDataDetector(types: types.rawValue)
+        } catch {
+            detector = nil
+            assertionFailure("Data detector failed to be created in \(self)")
+        }
+    }
+
+    /// Checks if the provided text contains links or not.
+    public func hasLinks(in text: String) -> Bool {
+        firstLink(in: text) != nil
+    }
+
+    /// Parses the first link of the provided text.
+    public func firstLink(in text: String) -> TextLink? {
+        let fullRange = NSRange(location: 0, length: text.utf16.count)
```

Comment:
A short comment why we use `utf16.count` would be nice here.

Reply:
There is also this NSRange init available: `NSRange(text.startIndex..<text.endIndex, in: text)`